x-pack/plugins/spaces/server/lib/spaces_client.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import Boom from 'boom';
+import { omit } from 'lodash';
+import { isReservedSpace } from '../../common/is_reserved_space';
+
+export class SpacesClient {
+  private readonly authorization: any;
+  private readonly callWithRequestSavedObjectRepository: any;
+  private readonly internalSavedObjectRepository: any;
+  private readonly request: any;
+
+  constructor(
+    authorization: any,
+    callWithRequestSavedObjectRepository: any,
+    internalSavedObjectRepository: any,
+    request: any
+  ) {
+    this.authorization = authorization;
+    this.callWithRequestSavedObjectRepository = callWithRequestSavedObjectRepository;
+    this.internalSavedObjectRepository = internalSavedObjectRepository;
+    this.request = request;
+  }
+
+  public async getAll() {
+    if (this.useRbac()) {
+      const { saved_objects } = await this.internalSavedObjectRepository.find({
+        type: 'space',
+        page: 1,
+        perPage: 1000,
+      });
+
+      const spaces = saved_objects.map(this.transformSavedObjectToSpace);
+
+      const spaceIds = spaces.map((space: any) => space.id);
```

Comment:
Here (and elsewhere in this file), we can take advantage of the `Space` interface via 
```javascript
import { Space } from '../../common/model/space';
```